### PR TITLE
stm32f4discovery: use openocd for flashing

### PR DESCRIPTION
**Pull request by mehlis**

### CODE REVIEW

**Comment by LudwigOrtmann** [comment]
If the `ELF` was used - wouldn't we need to `strip` it for space saving? Thereby losing the debugging symbols or requiring yet another copy?

**Comment by gebart** [comment]
@LudwigOrtmann The debug symbols do not take up any space as far as I can tell, they are all located at 0x0 in section `.debug` or something similar. I believe that OpenOCD only loads sections given proper memory segments. Try running `readelf -l` on your binary and it will list which sections belong to which segment, no `.debug_xxx` sections belong to any segments on my binaries.

`objdump -p -h` will show the segments and their flags, I have not verified this but I think that only segments marked as `ALLOC` will take up any memory space and `LOAD` means it will be loaded onto the device when flashing, at least this is how I understand that it is supposed to work on a desktop system when loading a regular program.

See also: 
 - http://stackoverflow.com/questions/3281698/explaining-readelf-s-output
 - http://www.skyfree.org/linux/references/ELF_Format.pdf
 - http://wiki.osdev.org/ELF

**Comment by LudwigOrtmann** [comment]
@gebart Thanks for clearing this up!

**Comment by PeterKietzmann** [comment]
Is this a candidate for Hack'n'ACK this evening?

**Comment by haukepetersen** [comment]
I would say so!

File: boards/stm32f4discovery/dist/flash.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+openocd -f "${RIOTBOARD}/${BOARD}/dist/openocd.cfg" \
+    -c "init" \
+    -c "targets" \
+    -c "reset" \
+    -c "halt" \
+    -c "flash protect 0 0 11 off" \
+    -c "flash erase_address 0x08000000 0x40000" \
```
Comment (BytesGalore) [inline boards/stm32f4discovery/dist/flash.sh]: To fully erase the flash, please change to:
`-c "flash erase_address 0x08000000 0x100000" \`

File: boards/stm32f4discovery/dist/debug.sh [hunk 2]
```diff
@@ -1,4 +1,19 @@
 #!/bin/sh
 
-echo "Debugging $1"
-arm-none-eabi-gdb -tui -command=$1 $2
+openocd -f "${RIOTBOARD}/${BOARD}/dist/openocd.cfg" \
```
Comment (thomaseichinger) [inline boards/stm32f4discovery/dist/debug.sh]: For this and all other `*.sh`s, in current openocd versions there exists `board/stm32f4discovery.cfg`.

**Comment by haukepetersen** [comment]
Works. ACK and go.

**Pull request merged**